### PR TITLE
ref(hydration error): Refactor and shrink padding in the hydration error modal

### PR DESCRIPTION
**Pull request by ryan953**
This PR tidies up the hydration error diff modal and related components:
- we're shrinking padding, so there's more space for content
- i've moved a big descriptive paragraph into a tooltip. i believe people understand what's going on, at least after the first interaction
- more consistent use of tooltips related to the `Before` & `After` labels
- more consistent wrapper html in the 'Chooser' component where the tabs live, and it's children
- the 'give feedback' button is getsentry specific. so it's not visible in the screenshots from my dev machine, but it hasn't changed.

| Area | Before | After |
| --- | --- | --- |
| Issue Details | SCR-20241113-oexa | SCR-20241113-oeyi
| Slider Diff | before | after-slider
| Side-By-Side |  | after-side-by-side
| HTML Diff | | after-html

### CODE REVIEW

File: static/app/components/replays/diff/replayTextDiff.tsx [hunk 1]
```diff
@@ -28,10 +30,19 @@ export function ReplayTextDiff({replay, leftOffsetMs, rightOffsetMs}: Props) {
...
 
   return (
-    <Fragment>
+    <Container>
+      <StyledAlert type="info" showIcon>
+        {tct(
+          `The HTML Diff is currently in beta and has known issues (e.g. the ‘before’ is sometimes empty). We are exploring different options to replace this view, please see [link: this ticket] for more details and share your feedback.`,
+          {
+            link: (
+              <ExternalLink href={'https://github.com/getsentry/sentry/issues/80092'} />
```
Comment (billyvg) [inline static/app/components/replays/diff/replayTextDiff.tsx:39]: can it be 

```suggestion
              <ExternalLink href="https://github.com/getsentry/sentry/issues/80092" />
```

File: static/app/components/tabs/tabList.tsx [hunk 2]
```diff
@@ -251,7 +251,7 @@ function BaseTabList({
 const collectionFactory = (nodes: Iterable<Node<any>>) => new ListCollection(nodes);
 
 /**
- * To be used as a direct child of the <Tabs /> component. See example usage
+ * To be used as a direct child of the`<Tabs />` component. See example usage
```
Comment (billyvg) [inline static/app/components/tabs/tabList.tsx:254]:
```suggestion
 * To be used as a direct child of the `<Tabs />` component. See example usage
```

**Review by billyvg — APPROVED** [review]

File: static/app/components/replays/breadcrumbs/replayComparisonModal.tsx [hunk 3]
```diff
@@ -36,7 +39,24 @@ export default function ReplayComparisonModal({
     <OrganizationContext.Provider value={organization}>
       <Header closeButton>
         <ModalHeader>
-          <h4>{t('Hydration Error')}</h4>
+          <Title>
+            {t('Hydration Error')}
+            <Tooltip
+              isHoverable
+              title={tct(
+                'This modal helps with debugging hydration errors by diffing the dom before and after the app hydrated. [boldBefore:Before] refers to the html rendered on the server. [boldAfter:After] refers to the html rendered on the client. Read more about [link:resolving hydration errors]',
```
Comment (michellewzhang) [inline static/app/components/replays/breadcrumbs/replayComparisonModal.tsx:47]:
```suggestion
                'This modal helps with debugging hydration errors by diffing the DOM before and after the app hydrated. [boldBefore:Before] refers to the HTML rendered on the server. [boldAfter:After] refers to the HTML rendered on the client. Read more about [link:resolving hydration errors]',
```

File: static/app/components/replays/diff/replayTextDiff.tsx [hunk 4]
```diff
@@ -28,10 +30,19 @@ export function ReplayTextDiff({replay, leftOffsetMs, rightOffsetMs}: Props) {
   );
 
   return (
-    <Fragment>
+    <Container>
+      <StyledAlert type="info" showIcon>
+        {tct(
+          `The HTML Diff is currently in beta and has known issues (e.g. the ‘before’ is sometimes empty). We are exploring different options to replace this view, please see [link: this ticket] for more details and share your feedback.`,
```
Comment (michellewzhang) [inline static/app/components/replays/diff/replayTextDiff.tsx:36]:
```suggestion
          `The HTML Diff is currently in beta and has known issues (e.g. the 'Before' is sometimes empty). We are exploring different options to replace this view. Please see [link: this ticket] for more details and share your feedback.`,
```

**Pull request merged**